docs/esp32/tutorial/udp_multicast.rst

```diff
@@ -0,0 +1,141 @@
+.. _esp32_udp_multicast:
+
+UDP MULTICAST
+=============
+
+Use UDP MULTICAST connections to communicate between different networks.
+
+* udp_multicast_client.py::
+
+    import network, socket, errno, struct, time
+
+    # connect to AP
+    SSID = 'SSID'
+    PASSWORD = 'PASSWORD'
+
+    wlan = network.WLAN(network.STA_IF)
+    wlan.ifconfig(('172.16.55.55', '255.255.255.0', '172.16.55.1', '172.16.55.1'))  # class B network
+    wlan.active(True)
+    wlan.connect(SSID, PASSWORD)
+    while not wlan.isconnected():
+        pass
+
+    print("udp_multicast_client.py")
+    # send and receive
+
+    def inet_aton(str_addr):
+        return bytes(map(int, str_addr.split(".")))
+
+    #TIMEOUT = None  # block
+    #TIMEOUT = 5  # s
+    TIMEOUT = 0  # non blocking
+
+    IP = '224.0.0.111'  # MULTICAST
+    PORT = 5555
+    sockaddr = (IP, PORT)
+
+    wlan = network.WLAN(network.STA_IF)
+    ip = wlan.ifconfig()[0]
```

Comment:
Having variable names only differ in case (`IP`  and `ip`) might not be the most readable for an example.

Reply:
Renamed "IP" to "MULTICAST_IP" and
"ip" to "client_ip"/"server_ip"